e2e/wdio/core/tests/dialog.e2e-spec.ts

```diff
@@ -0,0 +1,676 @@
+        it('should check auto dismissal', () => {
+            const selfDismissingDialogCount = 3;
+
+            for (let i = 0; i < selfDismissingDialogCount; i++) {
+                openDialog(stateDialog, i);
+                pause(4500);
+
+                expect(doesItExist(dialog)).toBe(false);
+            }
+        });
+
+        it('check ability to dismiss alerts', () => {
+            const dialogCount = getElementArrayLength(stateDialog + button);
+
+            for (let i = 0; i < dialogCount; i++) {
+                openDialog(stateDialog, i);
+                closeDialog();
+
+                expect(doesItExist(dialog)).toBe(false);
+            }
+        });
+
+        it('check the loading icon', () => {
+            openDialog(stateDialog, 3);
+
+            expect(isElementDisplayed(dialog + busyIndicator)).toBe(true);
+            closeDialog();
+        });
+    });
+
+    describe('dialog configuration examples', function() {
+        it('should check dialog is draggable', () => {
+            if (!browserIsFirefox()) {
+                // dragAndDrop not working correctly on Saucelabs for Edge/Chrome
+                return;
+            }
+
+            scrollIntoView(configurationDialog);
+            openDialog(configurationDialog);
+            const dialogStartLocationX = Math.floor(getElementLocation(dialogContainer, 0, 'x'));
+            const dialogStartLocationY = Math.floor(getElementLocation(dialogContainer, 0, 'y'));
+
+            clickAndDragElement(dialogStartLocationX + 20, dialogStartLocationY + 10, dialogStartLocationX - 100, dialogStartLocationY - 50);
+
+            expect(Math.floor(getElementLocation(dialogContainer, 0, 'x'))).not.toBe(dialogStartLocationX);
+            expect(Math.floor(getElementLocation(dialogContainer, 0, 'y'))).not.toBe(dialogStartLocationY);
+            closeDialog();
+        });
+
+        it('should check dialog is resizeable', () => {
+            openDialog(configurationDialog, 1);
+
+            checkResizingDialog();
+
+            closeDialog();
+        });
+
+        it('should check dialog only closes with footer button click', () => {
+            openDialog(configurationDialog, 2);
+            clickWithOption(dialog, 0, 5000, { x: -100, y: -100 });
+
+            expect(doesItExist(dialog)).toBe(true);
+            closeDialog();
+
+            expect(doesItExist(dialog)).toBe(false);
+        });
+    });
+
+    describe('dialog positioning example', function() {
+        it('should check dialog in bottom right', () => {
+            openDialog(positionDialog);
+
+            expect(getAttributeByName(dialogContainer, styleAttribute)).toContain(bottomRightPosition);
+            closeDialog();
+        });
+    });
+
+    describe('mobile dialog example', function() {
+        it('should check mobile property', () => {
+            openDialog(mobileDialog);
+
+            expect(getAttributeByName(dialogContainer, classAttribute)).toContain(mobileProperty);
```

Comment:
let create function in wdio.ts for that.

Reply:
for what exactly?

Reply:
getAttributeByName(dialogContainer, classAttribute) ==> getElementClass() or similar

Reply:
per discussion, do refactoring for generic methods in a separate PR